header.php

```diff
@@ -23,53 +23,12 @@
 
 <body <?php body_class(); ?>>
 
-<?php wp_body_open(); ?>
-
-<div class="webpage">
-
-	<a href="#site-content" class="screen-reader-shortcut"><?php esc_html_e( 'Skip to content', 'jarvis' ); ?></a>
-
-	<header class="site-header" id="header" role="banner">
-
-		<div class="site-header-content container">
-
-			<?php the_custom_logo(); ?>
-
-			<div class="branding">
-
-<?php
-	if ( is_front_page() && ! is_paged() ) {
-?>
-				<h1 class="site-title">
-					<?php bloginfo( 'name' ); ?>
-				</h1>
-<?php
-	} else {
-?>
-				<p class="site-title">
-					<a href="<?php echo esc_url( home_url( '/' ) ); ?>" rel="home"><?php bloginfo( 'name' ); ?></a>
-				</p>
 <?php
-	}
-
-	// Get site description.
-	$description = get_bloginfo( 'description', 'display' );
-
-	if ( $description || is_customize_preview() ) {
-?>
-				<p class="site-description">
-					<?php echo $description; /* WPCS: xss ok. */ ?>
-				</p>
-<?php
-	}
-?>
-
-			</div>
-
-			<?php get_template_part( 'parts/navigation' ); ?>
-
-		</div>
-
-	</header>
-
-	<div class="container" id="site-content">
+if ( function_exists( 'wp_body_open' ) ) {
+	wp_body_open();
+} else {
+	do_action( 'wp_body_open' );
+}
+do_action( 'jarvis_before_header' );
```

Comment:
This is one of the things I don't understand about adding page build compat. Why are there so many filters? Can't the same thing be accomplished with one?

Reply:
It's so you can add items before the header (like a top nav) and then display the regular header. For example, I can create a header template, and then add a template part for breadcrumbs that will show right after the header.